src/CsvHelper/TypeConversion/TypeConverterOptionsCache.cs

```diff
@@ -13,6 +13,7 @@ namespace CsvHelper.TypeConversion
 	public class TypeConverterOptionsCache
 	{
 		private Dictionary<Type, TypeConverterOptions> typeConverterOptions = new Dictionary<Type, TypeConverterOptions>();
+		private readonly object dictionaryLock = new object();

```

Comment:
Shouldn't this be `static`?

Reply:
`typeConverterOptions` isn't static so we just need to use lock per `TypeConverterOptionsCache` instance.

Reply:
@JoshClose @InRedikaWB is correct. You would not want the lock to be static, even if the class itself was treated as a singleton.

Reply:
How about using a [ConcurrentDictionary](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2) instead of a `Dictionary` + a lock?

Reply:
@0xced The lock block would be more efficient than the concurrent dictionary

Reply:
Do you have benchmarks to back that claim? 😉

Reply:
Yes
```
|               Method |     Mean |     Error |    StdDev |
|--------------------- |---------:|----------:|----------:|
|                 Lock | 1.042 μs | 0.0783 μs | 0.2284 μs |
| ConcurrentDictionary | 2.324 μs | 0.0935 μs | 0.2741 μs |
```